src/main/java/org/killbill/billing/client/KillBillHttpClient.java

```diff
@@ -103,9 +117,16 @@ public KillBillHttpClient(final String kbServerUrl, final String username, final
 
 
         final AsyncHttpClientConfig.Builder cfg = new AsyncHttpClientConfig.Builder();
-        cfg.setConnectTimeout(DEFAULT_HTTP_TIMEOUT_SEC * 1000);
-        cfg.setRequestTimeout(DEFAULT_HTTP_TIMEOUT_SEC * 1000);
-        cfg.setReadTimeout(DEFAULT_HTTP_TIMEOUT_SEC * 1000);
+
+        if (connectTimeOut != null && readTimeOut != null && requestTimeout != null) {
```

Comment:
> I think it's a good idea that the user of the client can decided if they want to set them independently

Ok, let's change it to make the timeouts configurable independently then (i.e. 3 `null` checks).

> At the same time I am not sure about the meaning of the request timeout

For reference:

* [setConnectTimeout](https://github.com/AsyncHttpClient/async-http-client/blob/master/client/src/main/java/org/asynchttpclient/AsyncHttpClientConfig.java#L664-L672)
* [setRequestTimeout](https://github.com/AsyncHttpClient/async-http-client/blob/master/client/src/main/java/org/asynchttpclient/AsyncHttpClientConfig.java#L717-L725)
* [setReadTimeout](https://github.com/AsyncHttpClient/async-http-client/blob/master/client/src/main/java/org/asynchttpclient/AsyncHttpClientConfig.java#L691-L698)